app/grandchallenge/algorithms/models.py

```diff
@@ -983,11 +983,15 @@ def get_jobs_with_same_inputs(
         existing_jobs = (
             Job.objects.filter(**unique_kwargs)
             .annotate(
-                inputs_match_count=Count(
-                    "inputs", filter=Q(inputs__in=existing_civs)
-                )
+                input_count=Count("inputs", distinct=True),
+                input_match_count=Count(
+                    "inputs", filter=Q(inputs__in=existing_civs), distinct=True
+                ),
+            )
+            .filter(
+                input_count=input_interface_count,
+                input_match_count=input_interface_count,
```

Comment:
Oversight in my last PR. James, I think you hinted at this with your comment about partially overlapping inputs, but I didn't grasp it back then and wrote an incomplete test for it. This fixes it, and I also added to the test to cover the case I had missed. 

We use a very similar annotation and filter pattern in 5 different places in the codebase now. I tried to look for a way to re-use it, but since it's on 5 different models with sometimes different field names I didn't find a way to move the logic to one place and then re-use it everywhere. If you have suggestions, I'm all ears. Otherwise, just being careful in the future and good coverage tests are the way to go for now.